application/common/System.cpp

```diff
@@ -0,0 +1,152 @@
+#include "loom/script/loomscript.h"
+#include "loom/common/utils/utString.h"
+#include "loom/common/platform//platformThread.h"
+#include <iostream>
+#include <stdio.h>
+#include <stdlib.h>
+
+class System
+{
+private:
+	static void zeroBuffer(char *buff)
+	{
+		unsigned int stringLength = strlen(buff);
+
+		// We are using an unsigned int, so we can expect the value to roll over once we are finished with the loop
+		for (unsigned int i = stringLength - 1; i >= 0 && i < stringLength; i--)
+		{
+			buff[i] = 0;
+		}
+	}
+
+protected:
+	static const int BUFFER_LENGTH = 1024;
+	int id;
+	FILE* pipe;
+	char buffer[BUFFER_LENGTH];
+
+	static int __stdcall getData(void *param)
+	{
+		System* self = (System*)param;
+
+		// Zero out the buffer
+		zeroBuffer(self->buffer);
+
+		while (true/*!feof(self->pipe)*/) 
+		{
+			// Get the next character!
+			int recievedChar = fgetc(self->pipe);
```

Comment:
Hmm. Yes - if you want to check for EOF return value you must keep it an int. But you don't do that - you only compare to values in unsigned char range. It doesn't really matter - you can leave it as is.